### PR TITLE
[SMTChecker] Fix handling of function calls where the function identifier is nested in a tuple.

### LINKED ISSUE #10524: [SMTChecker] ICE in vector<smtutil::Expression> solidity::frontend::SMTEncoder::symbolicArguments(const solidity::frontend::FunctionCall &)
## Description

This contract:
```solidity
pragma experimental SMTChecker;
library L {
    struct S {
        uint256[] data;
    }
    function f(S memory _s) internal {
    }
}
contract C {
    using L for L.S;
    function f() public returns (uint256) {
        L.S memory x;
        (x.f)();
    }
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/formal/SMTEncoder.cpp(2551): Throw in function vector<smtutil::Expression> solidity::frontend::SMTEncoder::symbolicArguments(const solidity::frontend::FunctionCall &)
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

when compiled with solc.

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer,  a bit indirectly:  I took every test in my corpus I use to start fuzzing, and add the SMT checker pragma to any that lacked it, then ran triage and minimized the tests (so they aren't exactly the corpus tests + SMT checker pragma).

## Environment

- Compiler version:  0.7.6-develop.2020.11.30+commit.7e7a42c6.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

### PR DESCRIPTION
**Pull request by blishko**
Fixes #10524.

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_2.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	using L for L.S;
+	function f() public returns (uint256) {
+		L.S memory x;
+		(x.f)();
+	}
+}
+// ----
+// Warning 6321: (181-188): Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
+// Warning 6031: (211-214): Internal error: Expression undefined for SMT solver.
+// Warning 6031: (211-214): Internal error: Expression undefined for SMT solver.
```
Comment (blishko) [inline test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_2.sol]: This is an example from #10524.

File: test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_1.sol [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+		x = 0;
+		((inc))();
+		assert(x == 1); // should hold
+	}
+
+	function inc() internal returns (uint) {
+		require(x < 100);
+		return ++x;
+	}
+}
```
Comment (blishko) [inline test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_1.sol:15]: This is an example where SMTChecker reported false positive before.

File: test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_2.sol [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	}
+	function f(S memory _s) internal {
+	}
+}
+
+contract C {
+	using L for L.S;
+	function f() public returns (uint256) {
+		L.S memory x;
+		(x.f)();
```
Comment (leonardoalt) [inline test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_2.sol]: Maybe add a return value to `f` and use it in an assertion here?

Comment (blishko) [reply test/libsolidity/smtCheckerTests/functions/functions_identifier_nested_tuple_2.sol]: Updated.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**